IdentityCore/src/oauth2/aad_base/MSIDClientInfo.m

```diff
@@ -53,4 +53,49 @@ - (NSString *)userIdentifier
     return [NSString stringWithFormat:@"%@.%@", self.uid, self.utid];
 }
 
+- (BOOL)isEqualToClientInfo:(MSIDClientInfo *)clientInfo
+{
+    if (!clientInfo)
+    {
+        return NO;
+    }
+    
+    BOOL result = YES;
+    result &= (!_json && !clientInfo->_json) || [_json isEqualToDictionary:clientInfo->_json];
```

Comment:
nit: this could be in NSJSONObject as I assume it might be necessary in multiple other places